annotations/views.py

```diff
@@ -1040,25 +1040,58 @@ def relation_details(request, source_concept_id, target_concept_id):
 def concept_details(request, conceptid):
     concept = get_object_or_404(Concept, pk=conceptid)
     appellations = Appellation.objects.filter(interpretation_id=conceptid)
-    texts = set()
-    appellations_by_text = OrderedDict()
-    for appellation in appellations:
-        text = appellation.occursIn
-        texts.add(text)
-        if text.id not in appellations_by_text:
-            appellations_by_text[text.id] = []
-        appellations_by_text[text.id].append(appellation)
-
-    template = loader.get_template('annotations/concept_details.html')
-    context = RequestContext(request, {
-        'user': request.user,
-        'concept': concept,
-        'appellations': appellations_by_text,
-        'texts': texts,
-    })
-
-    return HttpResponse(template.render(context))
 
+    response_format = request.GET.get('format', None)
+    if response_format != 'json':
+        texts = set()
+        appellations_by_text = OrderedDict()
+        for appellation in appellations:
+            text = appellation.occursIn
+            texts.add(text)
+            if text.id not in appellations_by_text:
+                appellations_by_text[text.id] = []
+            appellations_by_text[text.id].append(appellation)
+
+        template = loader.get_template('annotations/concept_details.html')
+        context = RequestContext(request, {
+            'user': request.user,
+            'concept': concept,
+            'appellations': appellations_by_text,
+            'texts': texts,
+        })
+        return HttpResponse(template.render(context))
+    else:
+        response = dict()
+        response['concept'] = concept
+        json = []
+        appellations_by_text = dict()
+        text = ""
+        for appellation in appellations:
```

Comment:
You might consider using [``values()``](https://docs.djangoproject.com/en/1.9/ref/models/querysets/#values) or [``values_list()``](https://docs.djangoproject.com/en/1.9/ref/models/querysets/#values-list) here, for performance' sake.